entity-framework/core/querying/related-data.md

```diff
@@ -109,6 +109,27 @@ Contents of `School` navigation of all People who are Students can be eagerly lo
   context.People.Include("School").ToList()
   ```
 
+### Filtered include
+
+> [!NOTE]
+> This feature was introduced in EF Core 5.0.
+
+Filtered include feature enables applying some operations directly on the included collection navigation, which allows for filtering and sorting of the results.
+
+Supported operations are: `Where`, `OrderBy`, `OrderByDescending`, `ThenBy`, `ThenByDescending`, `Skip`, and `Take`.
+
+Those operations should be applied inside the Include method:
+
+[!code-csharp[Main](../../../samples/core/Querying/RelatedData/Sample.cs#FilteredInclude)]
+
+Each included navigation allows only one unique set of filter operations. In case when Include operation is applied more than once on a given collection navigation (`blog.Posts` in the examples below), filter operations should only be applied on one of them. 
```

Comment:
```suggestion
Each included navigation allows only one unique set of filter operations. In cases where multiple Include operations are applied on a given collection navigation (`blog.Posts` in the examples below), filter operations must only be applied to one of them. 
```

Reply:
BTW what happens if I accidentally put non-identical operations, do we throw informatively?

Reply:
Non-identical filter we throw.
It should be can rather than should/must. If you use same collection multiple time and specify same filter then it works.

Reply:
Yeah, there's a sentence saying that below. Maybe it should be merged, I agree it's not totally clear as-is.

Reply:
we throw a decent error:
Different filters: '{filter1}' and '{filter2}' have been applied on the same included navigation. Only one unique filter per navigation is allowed. For more information on including related data, see http://go.microsoft.com/fwlink/?LinkID=746393.

Reply:
I anticipate main use will be to apply it only once and use regular include for all other instances. The case when u apply the same filter multiple times is probably only useful in the code generation scenarios or maybe copy-paste, brain-dead coding. I'm happy to merge the paragraphs together, but i think the second scenario is definitely lower priority